aiohttp/client_exceptions.py

```diff
@@ -248,17 +249,34 @@ class InvalidURL(ClientError, ValueError):
 
     # Derive from ValueError for backward compatibility
 
-    def __init__(self, url: Any) -> None:
+    def __init__(self, url: StrOrURL, description: Union[str, None] = None) -> None:
         # The type of url is not yarl.URL because the exception can be raised
         # on URL(url) call
-        super().__init__(url)
+        if description:
+            super().__init__(url, description)
+        else:
+            super().__init__(url)
+
+    @property
+    def url(self) -> StrOrURL:
+        return self.args[0]  # type: ignore[no-any-return]
 
     @property
-    def url(self) -> Any:
-        return self.args[0]
+    def description(self) -> Optional[str]:
```

Comment:
Let's avoid `Optional`:
```suggestion
    def description(self) -> "str | None":
```